From c08d9752ad3305f0f84d44607077654704d77384 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Tue, 29 Nov 2022 09:32:35 +0100
Subject: [PATCH 100/529] ARM: dts: stm32: deactivate the PWR regulator on SCMI
 for STM32MP15

The SCMI PWR regulator is only available for STM32MP13x
This reverts the commit:
Revert "ARM: dts: stm32: fix pwr regulators references to use scmi"

Signed-off-by: Pascal Paillet <p.paillet@st.com>
Change-Id: I52350b5d4a4c7ee37c7895647a0cc0c1300b57b9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/278089
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 arch/arm/boot/dts/stm32mp15-scmi.dtsi       | 51 ---------------------
 arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi |  2 +-
 arch/arm/boot/dts/stm32mp157c-dk2-scmi.dtsi |  2 +-
 arch/arm/boot/dts/stm32mp157c-ev1-scmi.dtsi |  2 +-
 4 files changed, 3 insertions(+), 54 deletions(-)

--- a/arch/arm/boot/dts/stm32mp15-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp15-scmi.dtsi
@@ -27,37 +27,6 @@
 				reg = <0x16>;
 				#reset-cells = <1>;
 			};
-
-			scmi_voltd: protocol@17 {
-				reg = <0x17>;
-
-				scmi_reguls: regulators {
-					#address-cells = <1>;
-					#size-cells = <0>;
-
-					scmi_reg11: reg11@0 {
-						reg = <0>;
-						regulator-name = "reg11";
-						regulator-min-microvolt = <1100000>;
-						regulator-max-microvolt = <1100000>;
-					};
-
-					scmi_reg18: reg18@1 {
-						voltd-name = "reg18";
-						reg = <1>;
-						regulator-name = "reg18";
-						regulator-min-microvolt = <1800000>;
-						regulator-max-microvolt = <1800000>;
-					};
-
-					scmi_usb33: usb33@2 {
-						reg = <2>;
-						regulator-name = "usb33";
-						regulator-min-microvolt = <3300000>;
-						regulator-max-microvolt = <3300000>;
-					};
-				};
-			};
 		};
 	};
 
@@ -77,26 +46,6 @@
 	};
 };
 
-&reg11 {
-	status = "disabled";
-};
-
-&reg18 {
-	status = "disabled";
-};
-
-&usb33 {
-	status = "disabled";
-};
-
-&usbotg_hs {
-	usb33d-supply = <&scmi_usb33>;
-};
-
-&usbphyc {
-	vdda1v1-supply = <&scmi_reg11>;
-	vdda1v8-supply = <&scmi_reg18>;
-};
 
 /delete-node/ &clk_hse;
 /delete-node/ &clk_hsi;
--- a/arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi
@@ -22,7 +22,7 @@
 };
 
 &dsi {
-	phy-dsi-supply = <&scmi_reg18>;
+	phy-dsi-supply = <&reg18>;
 	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
--- a/arch/arm/boot/dts/stm32mp157c-dk2-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157c-dk2-scmi.dtsi
@@ -29,7 +29,7 @@
 };
 
 &dsi {
-	phy-dsi-supply = <&scmi_reg18>;
+	phy-dsi-supply = <&reg18>;
 	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
--- a/arch/arm/boot/dts/stm32mp157c-ev1-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157c-ev1-scmi.dtsi
@@ -27,7 +27,7 @@
 };
 
 &dsi {
-	phy-dsi-supply = <&scmi_reg18>;
+	phy-dsi-supply = <&reg18>;
 	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
